app/workers/gps_report_worker.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+class InvalidSupplierError < StandardError; end
+
+class GPSReportWorker
+  include Sidekiq::Worker
+  include ActionView::Helpers::TextHelper
+
+  def perform
+    @date_range = Date.yesterday - 6.days..Date.yesterday
+    threads = suppliers.map { |supplier| Thread.new { GPSReport.new(@date_range, supplier).generate } }
+    post_results(threads.each(&:join).map(&:value))
+  end
+
+private
+
+  def suppliers
+    %w[geoamey serco]
+  end
```

Comment:
```suggestion
  SUPPLIERS = %w[geoamey serco]
```

I wonder if it's clearer to make this an explicit constant?